houston/generator/base.py

```diff
@@ -199,12 +204,36 @@ def record_outcome(self, mission, outcome):
         """
         self.__history.append(mission)
         self.__outcomes[mission] = outcome
+        if coverage:
+            self.__coverage[mission] = coverage
 
         if outcome.failed:
             self.__failures.add(mission)
 
 
     def generate(self, seed, resource_limits):
+        """
+        Generate missions and return them
+        """
+        assert isinstance(seed, int)
+
+        missions = []
+        self.prepare(seed, resource_limits)
+        stream = MissionGeneratorStream(self)
+        try:
+            self.__resource_usage = ResourceUsage()
+            self.__start_time = timeit.default_timer()
+            self.tick()
+            # TODO use threads
+            while True:
+                mission = stream.next()
+                missions.append(mission)
+        except StopIteration:
+            print("Done with generating missions")
+        return missions
+
+
+    def generate_and_run(self, seed, resource_limits, with_coverage=False, files_to_instrument={}):
```

Comment:
Tip: don't use lists or dictionaries as default arguments -- the same list/dict instance is shared by all invocations. Instead, use `None`, check if the argument is `None`, and if so, replace with a (freshly generated) default list/dict.

Reply:
@ChrisTimperley Why are you commenting on an old version?